org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/managers/EclipseBuildSupport.java

```diff
@@ -49,4 +58,30 @@ public boolean isBuildFile(IResource resource) {
 		return false;
 	}
 
+	@Override
+	public boolean fileChanged(IResource resource, CHANGE_TYPE changeType, IProgressMonitor monitor) throws CoreException {
+		if (resource == null || !applies(resource.getProject())) {
+			return false;
+		}
+		refresh(resource, changeType, monitor);
+		IProject project = resource.getProject();
+		if (ProjectUtils.isJavaProject(project)) {
+			IJavaProject javaProject = JavaCore.create(project);
+			IClasspathEntry[] classpath = javaProject.getRawClasspath();
+			for (IClasspathEntry entry : classpath) {
+				if (entry.getEntryKind() == IClasspathEntry.CPE_LIBRARY) {
+					IPath path = entry.getPath();
+					IFile r = ResourcesPlugin.getWorkspace().getRoot().getFile(path);
+					if (r != null && r.equals(resource)) {
+						IClasspathEntry[] rawClasspath = javaProject.getRawClasspath();
+						javaProject.setRawClasspath(new IClasspathEntry[0], monitor);
+						javaProject.setRawClasspath(rawClasspath, monitor);
```

Comment:
if the target classpathEntry is found, add `break` to exit ealier.

Reply:
updated.